plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts

```diff
@@ -67,36 +71,30 @@ export class DefaultCatalogCollatorFactory implements DocumentCollatorFactory {
     const {
       batchSize,
       discovery,
+      type,
       locationTemplate,
       filter,
       catalogClient,
       tokenManager,
+      entityTransformer,
     } = options;
 
+    this.type = type ?? 'software-catalog';
     this.locationTemplate =
       locationTemplate || '/catalog/:namespace/:kind/:name';
     this.filter = filter;
     this.batchSize = batchSize || 500;
     this.catalogClient =
       catalogClient || new CatalogClient({ discoveryApi: discovery });
     this.tokenManager = tokenManager;
+    this.entityTransformer =
```

Comment:
You may have already discussed this, but I would recommend a more specific name for this callback, something like `customizeCatalogDocument`, `extendCatalogDocument`, `getCustomCatalogDocumentFields`, or another name you might suggest. The entity transformer seems to be a specific function that transforms an entity into a partial search document, and the name "entityTransformer" sounds generic to me. I mean, I had to look through the code to understand what this method would return.